### DISCUSSION TITLE
Did anyone try using custom cache handler (via incremental cache handler path) in pages router?

### DISCUSSION THREAD

**Discussion by mskwon1** [vercel/next.js#57040] [Help] [answered]
### Summary

Hi I am trying to solve shared isr storage problem between multiple next js web server instances (in aws environment)

first I tried using efs but couldn't really make it to work :( 

and then I looked at custom cache handler in [app router docs](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath), although there aren't any sufficient documents, I tried redis custom cache handler and it is working like a charm. 

But I am worried about using this custom cache handler in pages router since it is still an exprimental option and nothing really is stated in the pages router documents. So.. is anyone using custom cache handler in pages router and had some kind of issues?

**Comment by better-salmon** [comment, marked as answer]
Hi, @mskwon1! Check out the [`@neshca/cache-handler`](https://caching-tools.github.io/next-shared-cache/) library. It supports both the App and the Pages router. And it simplifies custom configurations as well.

It is not a silver bullet, but I hope it will help you.

**Comment by JannikZed** [comment]
very old discussion! But I want to point out now our redis cache handler, that supports also  Next.js 16 with app / pages router and full cache components support: https://github.com/trieb-work/nextjs-turbo-redis-cache
